docs/reference-manual/pack.md

```diff
@@ -0,0 +1,83 @@
+---
+title: fn::pack
+---
+
+##### Defined in {style: "api", badge: "#include <functional/pack.hpp>"}
+
+---
+
+<!--
```

Comment:
@MykolaGolubyev 👋 
I tried to add some more stubs today and ran into this error.
Not sure what's going on here. Hope you can shed some light on this.

FWIW i can see `fn::pack` in the generated doxygen XML. Any pointers are welcome 🙇

Reply:
what's the error message?

Reply:
Copying the error out for visibility:

```
Exception in thread "main" java.lang.RuntimeException: 
markup parsing error:
    TOC item: reference-manual/pack
    full path: /Users/godexsoft/Development/functional/docs/reference-manual/pack.md
error handling include plugin <doxygen-doc>
  free param: fn::pack
  opts: {}
expected to find element <declname>
        at org.testingisdocumenting.znai.website.WebSite.throwParsingErrorMessage(WebSite.java:570)
        at org.testingisdocumenting.znai.website.WebSite.parseMarkupAndUpdateTocItemAndSearch(WebSite.java:562)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
        at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1117)
        at org.testingisdocumenting.znai.website.WebSite.parseMarkups(WebSite.java:501)
        at org.testingisdocumenting.znai.website.WebSite.parse(WebSite.java:193)
        at org.testingisdocumenting.znai.website.WebSite.parseAndDeploy(WebSite.java:179)
        at org.testingisdocumenting.znai.website.WebSite$Configuration.deployTo(WebSite.java:1034)
        at org.testingisdocumenting.znai.cli.ZnaiCliApp.generateDocs(ZnaiCliApp.java:197)
        at org.testingisdocumenting.znai.cli.ZnaiCliApp.start(ZnaiCliApp.java:99)
        at org.testingisdocumenting.znai.cli.ZnaiCliApp.start(ZnaiCliApp.java:70)
        at org.testingisdocumenting.znai.cli.ZnaiCliApp.main(ZnaiCliApp.java:75)
Caused by: java.lang.RuntimeException: error handling include plugin <doxygen-doc>
  free param: fn::pack
  opts: {}
expected to find element <declname>
        at org.testingisdocumenting.znai.parser.commonmark.MarkdownVisitor.handleIncludePlugin(MarkdownVisitor.java:232)
        at org.testingisdocumenting.znai.parser.commonmark.MarkdownVisitor.visit(MarkdownVisitor.java:155)
        at znaishaded.org.commonmark.node.CustomBlock.accept(CustomBlock.java:7)
        at znaishaded.org.commonmark.node.AbstractVisitor.visitChildren(AbstractVisitor.java:137)
        at znaishaded.org.commonmark.node.AbstractVisitor.visit(AbstractVisitor.java:28)
        at znaishaded.org.commonmark.node.Document.accept(Document.java:7)
        at org.testingisdocumenting.znai.parser.commonmark.MarkdownParser.parsePartial(MarkdownParser.java:98)
        at org.testingisdocumenting.znai.parser.commonmark.MarkdownParser.parse(MarkdownParser.java:66)
        at org.testingisdocumenting.znai.website.WebSite.parseMarkupAndUpdateTocItemAndSearch(WebSite.java:549)
        ... 10 more
Caused by: java.lang.IllegalArgumentException: expected to find element <declname>
        at org.testingisdocumenting.znai.utils.XmlUtils.lambda$nextLevelNodeByName$1(XmlUtils.java:87)
        at java.base/java.util.Optional.orElseThrow(Optional.java:403)
        at org.testingisdocumenting.znai.utils.XmlUtils.nextLevelNodeByName(XmlUtils.java:87)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenMemberParser.lambda$parseMember$0(DoxygenMemberParser.java:63)
        at java.base/java.util.stream.SpinedBuffer$1Splitr.forEachRemaining(SpinedBuffer.java:364)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:782)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenMemberParser.parseMember(DoxygenMemberParser.java:62)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenMemberParser.parseXml(DoxygenMemberParser.java:44)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenMemberParser.parse(DoxygenMemberParser.java:38)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenCompoundParser.parseMember(DoxygenCompoundParser.java:66)
        at java.base/java.util.stream.SpinedBuffer$1Splitr.forEachRemaining(SpinedBuffer.java:356)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:782)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenCompoundParser.parseXml(DoxygenCompoundParser.java:62)
        at org.testingisdocumenting.znai.doxygen.parser.DoxygenCompoundParser.parse(DoxygenCompoundParser.java:39)
        at org.testingisdocumenting.znai.doxygen.Doxygen.findAndParseCompound(Doxygen.java:136)
        at org.testingisdocumenting.znai.doxygen.Doxygen.getCachedOrFindAndParseCompound(Doxygen.java:101)
        at org.testingisdocumenting.znai.doxygen.plugin.DoxygenDocIncludePlugin.process(DoxygenDocIncludePlugin.java:58)
        at org.testingisdocumenting.znai.extensions.TrackingIncludePlugin.process(TrackingIncludePlugin.java:65)
        at org.testingisdocumenting.znai.parser.commonmark.MarkdownVisitor.handleIncludePlugin(MarkdownVisitor.java:228)
        ... 18 more
```

Reply:
I was able to reproduce and believe the problem is that znai expects `declname` for every argument:
```
        <param>
          <type>std::in_place_type_t&lt; T &gt;</type>
        </param>
        <param>
          <type>auto &amp;&amp;...</type>
          <declname>args</declname>
        </param>
```
As a temporary workaround (if possible) a to change signature so that Doxygen generates `declname`. 
Longer term, do you have a preference what to use as a default?

Reply:
I see. I'd go with empty string for unnamed params. I'm not sure how doxygen itself handles it though - i imagine we can't really doxy-comment on unnamed params. If some name is absolutely a must for znai then arg1, argN and args would do. or param1, paramN and params. Thats just my pref, i guess everyone will have their own

Reply:
Any progress with this issue @MykolaGolubyev ?
The tmp workaround of course works but i don't think anyone would want to sacrifice code for docs so better support for unnamed arguments is very much desired! 🙇‍♂️

Reply:
Thanks for reminding. Will allocate some time this week.

Reply:
https://github.com/testingisdocumenting/znai/pull/1212

Reply:
1.70 is released

Reply:
Thanks for letting me know 🙇 Will try as soon as i can.

Reply:
1.70 was released